core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java

```diff
@@ -380,10 +380,15 @@ public boolean delete(NameIdentifier ident, EntityType entityType, boolean casca
                               .build());
 
                   if (!cascade && !kvs.isEmpty()) {
+                    List<NameIdentifier> subEntities = Lists.newArrayListWithCapacity(kvs.size());
+                    for (Pair<byte[], byte[]> pair : kvs) {
+                      subEntities.add(entityKeyEncoder.decode(pair.getLeft()).getLeft());
+                    }
+
                     throw new NonEmptyEntityException(
                         String.format(
-                            "Entity %s has sub-entities, you should remove sub-entities first",
-                            ident));
+                            "Entity %s has sub-entities %s, you should remove sub-entities first",
+                            ident, subEntities));
```

Comment:
Will this `subEntities` be so large when there're so many entities? If so, is it necessary to print them all?

Reply:
No, I have limited the data retrieved from Gravitino, please see:

```java
  new KvRangeScan.KvRangeScanBuilder()
                              .start(directChild)
                              .end(endKey)
                              .startInclusive(true)
                              .endInclusive(false)
                              .limit(1)
                              .build());
```